### PR TITLE
Rename "master" branch to "devel"

### PR DESCRIPTION
**Pull request by nixpanic**
This PR renames all occasions of "master" to "devel" where the reference is relevant to Ceph-CSI.

Once this PR is merged, an admin of the GitHub project will need to rename the branch before next PRs can get merged. The GitHub documentation describes that all the existing PRs for the "master" branch will automatically be updated to the new "devel" branch.

Backport-of: #1890

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

**Review by Madhu-1 — APPROVED** [review]

**Comment by Madhu-1** [comment]
Once CI passes, This needs manual merge

**Comment by nixpanic** [comment]
**ci/centos/k8s-e2e-external-storage** probably does not work on release-v3.2, but it is an optional job in any case

**Comment by humblec** [comment]
Will wait for other tests to pass and merge :+1:

**Pull request merged**
